pkg/reconciler/scheduling/placement/placement_reconcile.go

```diff
@@ -97,18 +77,18 @@ func (c *controller) listLocations(clusterName logicalcluster.Name) ([]*scheduli
 	return ret, nil
 }
 
-func (c *controller) listWorkloadClusters(clusterName logicalcluster.Name) ([]*workloadv1alpha1.WorkloadCluster, error) {
-	items, err := c.workloadClusterIndexer.ByIndex(byWorkspace, clusterName.String())
+func (c *controller) listNamespace(clusterName logicalcluster.Name) ([]*corev1.Namespace, error) {
+	items, err := c.namespaceIndexer.ByIndex(byWorkspace, clusterName.String())
 	if err != nil {
 		return nil, err
 	}
-	ret := make([]*workloadv1alpha1.WorkloadCluster, 0, len(items))
+	ret := make([]*corev1.Namespace, 0, len(items))
 	for _, item := range items {
-		ret = append(ret, item.(*workloadv1alpha1.WorkloadCluster))
+		ns := item.(*corev1.Namespace)
+		_, foundPlacement := ns.Annotations[schedulingv1alpha1.PlacementAnnotationKey]
```

Comment:
would rather drop the annotation check here and let just the reconciler do the filtering.